src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1066 @@
+			if (e.keyCode === 27) {
+				if (tooltip.classList.contains('show')) {
+					tooltip.setAttribute('aria-hidden', 'true');
+					tooltip.classList.remove('show');
+				}
+			}
+			else if (e.keyCode === 13 || e.keyCode === 32) {
+				tooltip.setAttribute('aria-hidden', 'false');
+				tooltip.classList.toggle('show');
+				previewModalHeader.focus();
+			}
+		});
+
+		tooltip.addEventListener('keydown', (e: KeyboardEvent) => {
+			if (e.keyCode === 27) {
+				if (tooltip.classList.contains('show')) {
+					tooltip.setAttribute('aria-hidden', 'true');
+					tooltip.classList.remove('show');
+				}
+			}
+			else if (e.keyCode === 9) {
+				e.preventDefault();
+				if (e.target === previewModalBody) {
+					previewModalHeader.focus();
+				} else {
+					previewModalBody.focus();
+				}
+			}
+		});
+
+		window.addEventListener('click', (event) => {
+			const target = event.target as HTMLTextAreaElement;
+			if (!target.matches('.tooltip')) {
+				if (tooltip.classList.contains('show')) {
+					tooltip.classList.remove('show');
+				}
+			}
+		});
+	}
+
+
+	private createDropDown() {
+		const dropdownBtn = document.querySelector('#dropdown_btn');
+		const dropdown = document.querySelector('#dropdown') as HTMLInputElement;
+
+		dropdownBtn.addEventListener('click', () => {
+			dropdown.classList.toggle('show');
+		});
+
+		dropdownBtn.addEventListener('keydown', (e: KeyboardEvent) => {
+			if (e.keyCode === 13 || e.keyCode === 32) {
+				const dropdownFirstElement = document.querySelector('#dropdown').firstElementChild.children[0] as HTMLInputElement;
+				dropdown.classList.toggle('show');
+				dropdownFirstElement.focus();
+			}
+		});
+
+		dropdown.addEventListener('keydown', (e: KeyboardEvent) => {
+			if (e.keyCode === 27) {
+				if (dropdown.classList.contains('show')) {
+					dropdown.classList.remove('show');
+					const currentSelection = document.querySelector('.move:focus') as HTMLInputElement;
+					currentSelection.blur();
+				}
+			}
+		});
+
+		const body = document.querySelector('body');
+
+		if (body.classList.contains('windows') || body.classList.contains('linux')) {
+			const macOnly = document.querySelector('#dropdown_mac-only');
+			macOnly.remove();
+		} else if (body.classList.contains('mac')) {
+			const windowsLinuxOnly = document.querySelector('#dropdown_windows_linux-only');
+			windowsLinuxOnly.remove();
+		}
+
+		window.addEventListener('click', (event) => {
+			const target = event.target as HTMLTextAreaElement;
+			if (!target.matches('.dropdown')) {
+				if (dropdown.classList.contains('show')) {
+					dropdown.classList.remove('show');
+				}
+			}
+		});
+
+		dropdown.addEventListener('keydown', function (e: KeyboardEvent) {
+			const dropdownLastElement = document.querySelector('#dropdown').lastElementChild.children[0] as HTMLInputElement;
+			const dropdownFirstElement = document.querySelector('#dropdown').firstElementChild.children[0] as HTMLInputElement;
+			if (e.keyCode === 9) {
```

Comment:
keycodes, these are just magic numbers and should be using a constants class or something

Reply:
These have been added to a class